App/src/pages/CourseDetails.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useEffect, useState } from "react";
+import { useLocation } from "react-router-dom";
+import { Card, Heading, Text } from "@chakra-ui/react";
+
+function useQuery() {
+    return new URLSearchParams(useLocation().search);
+}
+
+const CourseDetails = () => {
+    const query = useQuery(); // Use the useQuery function
+
+    // Access specific query parameters
+    const subject: string = query.get("subject");
+    const catalogNumber: string = query.get("catalog");
+    const [data, setData] = useState([]);
+
+    useEffect(() => {
+        fetch("/data.json")
```

Comment:
This actually always crashes, because there is no data.json file. The endpoint to get course info based on the subject/catalog already exists though, so you can just fetch that.

Reply:
now using requestCourseSettings() which follows the recommended pattern